### PR TITLE
Added PIC18F4420-xP and PIC18F4520-xP

### PR DESCRIPTION
**Pull request by Misca1234**
Adding microchips MCU
PIC18F4420 and PIC184520 40 pin DIP variant

(this a re push of 
https://github.com/KiCad/kicad-symbols/pull/328
in smaller pieces)

http://ww1.microchip.com/downloads/en/DeviceDoc/39631E.pdf

bild

------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] An example screenshot image is very helpful
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required

### CODE REVIEW

**Comment by Misca1234** [comment]
Changed RE3 to input 

bild

**Comment by evanshultz** [comment]
Since pin 1 can be a GPIO, I think it should stay Bidirectional type.

@fauxpark 
You can probably spot anything that is non-conforming now real quick. Would you mind taking a look?

File: MCU_Microchip_PIC18.dcm [hunk 1]
```diff
@@ -252,6 +252,12 @@ K Flash-Based 8-Bit Microcontroller
 F http://ww1.microchip.com/downloads/en/DeviceDoc/39564c.pdf
 $ENDCMP
 #
+$CMP PIC18F4420-xP
+D 16K Flash, 768b SRAM, 256b EEPROM, ADC, DIP-40
```
Comment (fauxpark) [inline MCU_Microchip_PIC18.dcm]: The SRAM and EEPROM sizes are in bytes, so they should be uppercase here.

File: MCU_Microchip_PIC18.dcm [hunk 2]
```diff
@@ -252,6 +252,12 @@ K Flash-Based 8-Bit Microcontroller
...
+K 16K Flash, 768b SRAM, 256b EEPROM, ADC
```
Comment (fauxpark) [inline MCU_Microchip_PIC18.dcm]: The keywords field should be a little more informative than just the description again. Seems like the usual thing for this lib is "Flash-Based 8-Bit Microcontroller"

File: MCU_Microchip_PIC18.dcm [hunk 3]
```diff
@@ -354,6 +360,12 @@ K Flash-Based 8-Bit Microcontroller
 F http://ww1.microchip.com/downloads/en/DeviceDoc/39564c.pdf
 $ENDCMP
 #
+$CMP PIC18F4520-xP
+D 32K Flash, 1536b SRAM, 256b EEPROM, ADC, DIP-40
+K 32K Flash, 1536b SRAM, 256b EEPROM, ADC
```
Comment (fauxpark) [inline MCU_Microchip_PIC18.dcm]: And again for these two lines.

**Comment by fauxpark** [comment]
Hah, I know even less about PICs than AVRs, but other than the description and keywords this looks good to me.

The datasheet has this to say about RE3 (10.5, p117):

> The fourth pin of PORTE (MCLR/VPP/RE3) is an input only pin. Its operation is controlled by the MCLRE Configuration bit. When selected as a port pin (MCLRE = 0), it functions as a digital input only pin; as such, it does not have TRIS or LAT bits associated with its operation.

So I think it is correct to have it as Input.

**Comment by evanshultz** [comment]
@fauxpark 
Got it. Thanks! I figured your work would help to be a template for all MCUs.

@Misca1234 
If you can update the description, and the other PRs you have for the other packages of this MCU, this will be ready to merge and I'll move on to the other PRs.

**Comment by Misca1234** [comment]
Keyword updated

**Comment by fauxpark** [comment]
@Misca1234 could you also do the same for the 4520?

**Comment by evanshultz** [comment]
Dah. Too fast. I didn't realize what the above was asking for until it was too late. Since there are other open PRs for this family I'll request it there.

**Pull request merged**